enhancements/sig-policy/62-namespace-labelselector/README.md

```diff
@@ -0,0 +1,199 @@
+# Core Policy Module
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Implement `MatchLabels` and `MatchExpressions` (collectively known as `LabelSelector`) for
+`ConfigurationPolicies` in order to further filter the list of namespaces to which an object should
+be applied.
+
+## Motivation
+
+Currently the `ConfigurationPolicy` has a `spec.namespaceSelector` that includes `include` and
+`exclude` arrays of strings that optionally accept a leading and/or trailing wildcard to filter by
+the name of the namespace. In order to more closely align with Kubernetes use cases, the proposal is
+to implement a `LabelSelector` along with these arrays in order to filter namespaces by their label.
+The `LabelSelector` struct includes `MatchLabels` and `MatchExpressions`. In addition to aligning
+with Kubernetes practices, this will cut down on processing the namespace list for clusters with a
+large number of namespaces since a shorter list would be returned from the Kubernetes API.
+
+Futhermore, an additional proposal is to use the `filepath` Go library rather than a custom wildcard
+implementation for the `include` and `exclude` lists. This will bring maintainability and additional
+power to the namespace filtering. This will not be a breaking change since the current behavior is
+to only support leading and trailing wildcards, and a pattern like `foo*bar` is treated as a literal
+string, which is not a valid namespace name.
+
+### Goals
+
+1. Filter the namespace list by `LabelSelector` for the `NamespaceSelector`
+2. Allow `LabelSelector` and `include`/`exclude` arrays to operate together when both are provided
+   or separately when provided individually
+3. Replace custom wildcard code with `filepath`
```

Comment:
To clarify, this could link to https://pkg.go.dev/path/filepath#Match, which documents the rules the wildcards would follow.